datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/ingest/source/UpsertIngestionSourceResolver.java

```diff
@@ -119,4 +131,19 @@ private DataHubIngestionSourceSchedule mapSchedule(final UpdateIngestionSourceSc
     result.setTimezone(input.getTimezone());
     return result;
   }
+
+  private String optionallySetPipelineName(String recipe, String ingestionSourceUrn) {
+    try {
+      JSONObject jsonRecipe = new JSONObject(recipe);
+      boolean hasPipelineName = jsonRecipe.has("pipeline_name") && jsonRecipe.get("pipeline_name") != null && !jsonRecipe.get("pipeline_name").equals("");
```

Comment:
Feel free to punt: I'd store "pipeline_name" in a constant since we reuse it 4 times as a raw string here!